TestIngesterTransfer fails intermittenly

Test that relies on system clock in intermittent failing shocker. News at 9pm.

c.f. https://circleci.com/gh/weaveworks/cortex/910

```
--- FAIL: TestIngesterTransfer (6.01s)

	Error Trace:	ingester_lifecycle_test.go:155

	Error:		Not equal: &cortex.QueryResponse{Timeseries: []cortex.TimeSeries{cortex.TimeSeries{Labels:[]cortex.LabelPair{cortex.LabelPair{Name:wire.Bytes{0x5f, 0x5f, 0x6e, 0x61, 0x6d, 0x65, 0x5f, 0x5f}, Value:wire.Bytes{0x66, 0x6f, 0x6f}}}, Samples:[]cortex.Sample{cortex.Sample{Value:456, TimestampMs:123000}}}},

			} (expected)

			        != &cortex.QueryResponse{} (actual)

			

			Diff:

			--- Expected

			+++ Actual

			@@ -1,2 +1,2 @@

			-(*cortex.QueryResponse)(0xc4202a21a0)(&QueryResponse{Timeseries:[{[{[95 95 110 97 109 101 95 95] [102 111 111]}] [{456 123000}]}],})

			+(*cortex.QueryResponse)(0xc4202a2180)(&QueryResponse{Timeseries:[],})

			 
```

--- a/pkg/ingester/ingester_claim.go
+++ b/pkg/ingester/ingester_claim.go
@@ -98,9 +98,6 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 		sentChunks.Add(float64(len(descs)))
 	}
 
-	if err := stream.SendAndClose(&client.TransferChunksResponse{}); err != nil {
-		return err
-	}
 
 	if err := i.ClaimTokensFor(fromIngesterID); err != nil {
 		return err
@@ -114,7 +111,9 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 	}
 	i.userStates = userStates
 
-	return nil
+	// Close the stream last, as this is what tells the "from" ingester that
+	// it's OK to shut down.
+	return stream.SendAndClose(&client.TransferChunksResponse{})
 }
 
 func toWireChunks(descs []*desc) ([]client.Chunk, error) {

--- a/pkg/ingester/ingester_claim.go
+++ b/pkg/ingester/ingester_claim.go
@@ -98,7 +98,6 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 		sentChunks.Add(float64(len(descs)))
 	}
 
-
 	if err := i.ClaimTokensFor(fromIngesterID); err != nil {
 		return err
 	}

--- a/pkg/ingester/ingester_claim.go
+++ b/pkg/ingester/ingester_claim.go
@@ -112,7 +112,12 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 
 	// Close the stream last, as this is what tells the "from" ingester that
 	// it's OK to shut down.
-	return stream.SendAndClose(&client.TransferChunksResponse{})
+	if err := stream.SendAndClose(&client.TransferChunksResponse{}); err != nil {
+		level.Error(util.Logger).Log("msg", "Error closing TransferChunks stream", "ingester", fromIngesterID, "err", err)
+		return err
+	}
+	level.Info(util.Logger).Log("msg", "Successfully transferred chunks to ingester", "ingester", fromIngesterID)
+	return nil
 }
 
 func toWireChunks(descs []*desc) ([]client.Chunk, error) {

--- a/pkg/ingester/ingester_lifecycle.go
+++ b/pkg/ingester/ingester_lifecycle.go
@@ -425,11 +425,12 @@ func (i *Ingester) findTargetIngester() (*ring.IngesterDesc, error) {
 	for {
 		ingester, err := findIngester()
 		if err != nil {
-			level.Error(util.Logger).Log("msg", "Error looking for pending ingester: %v", err)
+			level.Debug(util.Logger).Log("msg", "Error looking for pending ingester", "err", err)
 			if time.Now().Before(deadline) {
 				time.Sleep(i.cfg.SearchPendingFor / pendingSearchIterations)
 				continue
 			} else {
+				level.Warn(util.Logger).Log("msg", "Could not find pending ingester before deadline", "err", err)
 				return nil, err
 			}
 		}

--- a/pkg/ring/consul_client.go
+++ b/pkg/ring/consul_client.go
@@ -178,11 +178,11 @@ func (c *consulClient) CAS(key string, f CASCallback) error {
 			ModifyIndex: index,
 		}, writeOptions)
 		if err != nil {
-			level.Error(util.Logger).Log("msg", "error CASing", "ley", key, "err", err)
+			level.Error(util.Logger).Log("msg", "error CASing", "key", key, "err", err)
 			continue
 		}
 		if !ok {
-			level.Error(util.Logger).Log("msg", "error CASing, trying again", "key", key, "index", index)
+			level.Debug(util.Logger).Log("msg", "error CASing, trying again", "key", key, "index", index)
 			continue
 		}
 		return nil

